src/ast/irbuilderbpf.cpp

```diff
@@ -1340,5 +1355,82 @@ StoreInst *IRBuilderBPF::createAlignedStore(Value *val,
 #endif
 }
 
+void IRBuilderBPF::CreateProbeRead(Value *ctx,
+                                   Value *dst,
+                                   const SizedType &type,
+                                   Value *src,
+                                   AddrSpace as,
```

Comment:
We don't need to take address space as an argument since it can just be derived from type (and that's what's happening at every call site now anyway). Same for `CreateLoad`

Reply:
In `probereadDatastructElem()`, the type passed to `CreateProbeRead()` is `elem_type`, but `AddrSpace` comes from `data_type`. Let me verify that both have the same `AddrSpace`.

Reply:
There are currently cases where the type's `AddrSpace` isn't set (e.g. fields of a struct have it set to `none`, while the struct's AS is `user` or `kernel`) and we need to pass in a different AS; for now I've kept the extra argument, but made it optional (defaults to `type.GetAS()`) and added a comment.